src/prefect/backend/flow_run.py

```diff
@@ -735,19 +738,25 @@ def get_all_task_runs(self) -> List["TaskRunView"]:
                 "Please load the tasks you are interested in individually."
             )
 
+        # Freeze the cached task runs we will return to prevent race conditions if
+        # this object is retrieving task runs in multiple threads
+        cached_task_runs = list(self._cached_task_runs.values())
+        cached_task_run_ids = {task_run.task_run_id for task_run in cached_task_runs}
+
```

Comment:
@bunchesofdonald A new change here that prevents a race condition if `self._cached_task_runs` is updated by another thread while the query is running. We wouldn't include those task runs in the `nin` call so they'd be returned by the API then on L759 we'd append the cached runs resulting in duplicates.